src/data/biomes/cave.ts

```diff
@@ -0,0 +1,134 @@
+      SpeciesId.PARAS,
+      SpeciesId.TEDDIURSA,
+      SpeciesId.WHISMUR,
+      SpeciesId.ROGGENROLA,
+      SpeciesId.WOOBAT,
+      SpeciesId.DIGGERSBY,
+      SpeciesId.NACLI,
+    ],
+  },
+  [BiomePoolTier.UNCOMMON]: {
+    [TimeOfDay.DAWN]: [],
+    [TimeOfDay.DAY]: [],
+    [TimeOfDay.DUSK]: [SpeciesId.ROCKRUFF],
+    [TimeOfDay.NIGHT]: [],
+    [TimeOfDay.ALL]: [SpeciesId.GEODUDE, SpeciesId.MAKUHITA, SpeciesId.NOSEPASS, SpeciesId.NOIBAT, SpeciesId.WIMPOD],
+  },
+  [BiomePoolTier.RARE]: {
+    [TimeOfDay.DAWN]: [],
+    [TimeOfDay.DAY]: [],
+    [TimeOfDay.DUSK]: [],
+    [TimeOfDay.NIGHT]: [],
+    [TimeOfDay.ALL]: [SpeciesId.ONIX, SpeciesId.FERROSEED, SpeciesId.CARBINK, SpeciesId.GLIMMET],
+  },
+  [BiomePoolTier.SUPER_RARE]: {
+    [TimeOfDay.DAWN]: [],
+    [TimeOfDay.DAY]: [],
+    [TimeOfDay.DUSK]: [],
+    [TimeOfDay.NIGHT]: [],
+    [TimeOfDay.ALL]: [SpeciesId.SHUCKLE],
+  },
+  [BiomePoolTier.ULTRA_RARE]: {
+    [TimeOfDay.DAWN]: [],
+    [TimeOfDay.DAY]: [],
+    [TimeOfDay.DUSK]: [],
+    [TimeOfDay.NIGHT]: [],
+    [TimeOfDay.ALL]: [SpeciesId.UXIE],
+  },
+  [BiomePoolTier.BOSS]: {
+    [TimeOfDay.DAWN]: [],
+    [TimeOfDay.DAY]: [],
+    [TimeOfDay.DUSK]: [],
+    [TimeOfDay.NIGHT]: [],
+    [TimeOfDay.ALL]: [
+      SpeciesId.PARASECT,
+      SpeciesId.ONIX,
+      SpeciesId.CROBAT,
+      SpeciesId.URSARING,
+      SpeciesId.EXPLOUD,
+      SpeciesId.PROBOPASS,
+      SpeciesId.GIGALITH,
+      SpeciesId.SWOOBAT,
+      SpeciesId.DIGGERSBY,
+      SpeciesId.NOIVERN,
+      SpeciesId.GOLISOPOD,
+      SpeciesId.GARGANACL,
+    ],
+  },
+  [BiomePoolTier.BOSS_RARE]: {
+    [TimeOfDay.DAWN]: [],
+    [TimeOfDay.DAY]: [],
+    [TimeOfDay.DUSK]: [SpeciesId.LYCANROC],
+    [TimeOfDay.NIGHT]: [],
+    [TimeOfDay.ALL]: [SpeciesId.SHUCKLE, SpeciesId.FERROTHORN, SpeciesId.GLIMMORA],
+  },
+  [BiomePoolTier.BOSS_SUPER_RARE]: {
+    [TimeOfDay.DAWN]: [],
+    [TimeOfDay.DAY]: [],
+    [TimeOfDay.DUSK]: [],
+    [TimeOfDay.NIGHT]: [],
+    [TimeOfDay.ALL]: [SpeciesId.UXIE],
+  },
+  [BiomePoolTier.BOSS_ULTRA_RARE]: {
+    [TimeOfDay.DAWN]: [],
+    [TimeOfDay.DAY]: [],
+    [TimeOfDay.DUSK]: [],
+    [TimeOfDay.NIGHT]: [],
+    [TimeOfDay.ALL]: [SpeciesId.TERAPAGOS],
+  },
+};
+
+const trainerPool = {
+  [BiomePoolTier.COMMON]: [TrainerType.BACKPACKER, TrainerType.HIKER],
+  [BiomePoolTier.UNCOMMON]: [TrainerType.ACE_TRAINER, TrainerType.BLACK_BELT],
+  [BiomePoolTier.RARE]: [],
+  [BiomePoolTier.SUPER_RARE]: [],
+  [BiomePoolTier.ULTRA_RARE]: [],
+  [BiomePoolTier.BOSS]: [TrainerType.BROCK, TrainerType.ROXANNE, TrainerType.ROARK],
+  [BiomePoolTier.BOSS_RARE]: [],
+  [BiomePoolTier.BOSS_SUPER_RARE]: [],
+  [BiomePoolTier.BOSS_ULTRA_RARE]: [],
+};
+
+const weatherPool = {
+  [WeatherType.NONE]: 1,
+  [WeatherType.SUNNY]: 0,
+  [WeatherType.RAIN]: 0,
+  [WeatherType.SANDSTORM]: 0,
+  [WeatherType.HAIL]: 0,
+  [WeatherType.SNOW]: 0,
+  [WeatherType.FOG]: 0,
+  [WeatherType.HEAVY_RAIN]: 0,
+  [WeatherType.HARSH_SUN]: 1,
```

Comment:
```suggestion
  [WeatherType.HARSH_SUN]: 0,
```
Not sure why a cave would have Desolate Land weather...

Reply:
That's a typo lol